conda_smithy/cli.py

```diff
@@ -179,16 +179,50 @@ def __init__(self, parser):
         scp = self.subcommand_parser
         scp.add_argument("--feedstock_directory", default=os.getcwd(),
                          help="The directory of the feedstock git repository.")
+        scp.add_argument("-c", "--commit", nargs='?', choices=["edit", "auto"], const="edit",
+                         help="Whether to setup a commit or not.")
 
     def __call__(self, args):
         try:
             configure_feedstock.main(args.feedstock_directory)
             print("\nCI support files regenerated. These need to be pushed to github!")
```

Comment:
Fixed below. Now only states to push if there were changes. Also if there were no changes, notes that everything is up-to-date.